master/buildbot/test/unit/test_status_stash.py

```diff
@@ -34,26 +39,71 @@ def setUp(self):
     def tearDown(self):
         pass
 
-    def test_buildStarted_sends_inprogress(self):
+    def test_buildStarted_sends_inprogress(self, mock_send):
         """
         buildStarted should send INPROGRESS
         """
-        self.status.send = Mock()
         self.status.buildStarted('fakeBuilderName', self.build)
-        self.status.send.assert_called_with('fakeBuilderName', self.build, INPROGRESS)
+        mock_send.assert_called_with('fakeBuilderName', self.build, INPROGRESS)
 
-    def test_buildFinished_sends_successful_on_success(self):
+    def test_buildFinished_sends_successful_on_success(self, mock_send):
         """
         buildFinished should send SUCCESSFUL if called with SUCCESS
         """
-        self.status.send = Mock()
         self.status.buildFinished('fakeBuilderName', self.build, SUCCESS)
-        self.status.send.assert_called_with('fakeBuilderName', self.build, SUCCESSFUL)
+        mock_send.assert_called_with('fakeBuilderName', self.build, SUCCESSFUL)
 
-    def test_buildFinished_sends_failed_on_failure(self):
+    def test_buildFinished_sends_failed_on_failure(self, mock_send):
         """
         buildFinished should send FAILED if called with anything other than SUCCESS
         """
-        self.status.send = Mock()
         self.status.buildFinished('fakeBuilderName', self.build, FAILURE)
-        self.status.send.assert_called_with('fakeBuilderName', self.build, FAILED)
+        mock_send.assert_called_with('fakeBuilderName', self.build, FAILED)
+
+
+@patch.object(StashStatusPush, '_send')
```

Comment:
we usually prefer to use patch in setup

in setUp(self):

self.status._send = mock.Mock()